contrib/world_builder/include/world_builder/point.h

```diff
@@ -0,0 +1,167 @@
+
+namespace WorldBuilder
+{
+  template<int dim>
+  class Point
+  {
+    public:
+      /**
+       * Constructor. Constructs a Point at (0,0) in 2d or (0,0,0) in 3d
+       * with a Cartesian coordinate system.
+       */
+      Point();
+
+      /**
+       * Constructor. Constructs a Point from a std::array<double,dim> and
+       * a coordinate system which is set by default to Cartesian.
+       */
+      Point(const std::array<double,dim> &array, CoordinateSystem coordinate_system = CoordinateSystem::cartesian);
+
+      /**
+       * Constructor. Constructs a Point from an other Point and
+       * a coordinate system which is set by default to Cartesian.
+       */
+      Point(const Point<dim> &point, CoordinateSystem coordinate_system = CoordinateSystem::cartesian);
+
+      /**
+       * Constructor. Constructs a 2d Point from two doubles and
+       * a coordinate system which is set by default to Cartesian.
+       */
+      Point(const double x, const double y, CoordinateSystem coordinate_system = CoordinateSystem::cartesian);
+
+      /**
+       * Constructor. Constructs a 3d Point from three doubles and
+       * a coordinate system which is set by default to Cartesian.
+       */
+      Point(const double x, const double y, const double z, CoordinateSystem coordinate_system = CoordinateSystem::cartesian);
+
+      /**
+       * Destructor
+       */
+      ~Point();
+
+      Point<dim> operator=(const Point<dim> &point);
+
+      /**
+       * dot product
+       */
+      double operator*(const Point<dim> &point) const;
+
+
+      /**
+       * Multiply the vector with a scalar
+       */
+      Point<dim> operator*(const double scalar) const;
+
+      /**
+       * add two points
+       */
+      Point<dim> operator+(const Point<dim> &point) const;
+
+
+      /**
+       * Substract two points
+       */
+      Point<dim> operator-(const Point<dim> &point) const;
+
+
+
+      /**
+       * Multiply the vector with a scalar
+       */
+      Point<dim> &operator*=(const double scalar);
+
+      /**
+       * add two points
+       */
+      Point<dim> &operator+=(const Point<dim> &point);
+
+
+      /**
+       * substract two points
+       */
+      Point<dim> &operator-=(const Point<dim> &point);
+
+      /**
+       * access index (const)
+       */
+      const double &operator[](const unsigned int index) const;
+
+
+      /**
+       * access index
+       */
+      double &operator[](const unsigned int index);
+
+      /**
+       * access index (const)
+       */
+      const double &operator()(const unsigned int index) const;
+
+      /**
+       * access index
+       */
+      double &operator()(const unsigned int index);
+
+
+      /**
+       * return the internal array which stores the point data.
+       */
+      std::array<double,dim> get_array() const;
+
+
+      /**
+       * returns the coordinate system associated with the data.
+       */
+      CoordinateSystem get_coordinate_system() const;
```

Comment:
const &?

Reply:
It is just an enum, I don't think that is necessary.